Breadcrumbs.py

```diff
@@ -1,94 +1,127 @@
 # -*- coding: utf-8 -*-
-import sublime, sublime_plugin
-import math
+import re
+import sublime
+import sublime_plugin
+
+
+if int(sublime.version()) > 3124:
+  import html
+  minihtml_available = True
+  viewevents_available = True
+else:
+  minihtml_available = False
+  viewevents_available = False
+
 
 try:
   xrange
 except NameError:
   xrange = range
 
+
 def get_tab_size(view):
   return int(view.settings().get('tab_size', 8))
 
-def get_row_indentation(points, view, tab_size, limit=1e20):
-  pos = 0
-  for pt in points:
-    ch = view.substr(pt)
 
+def get_row_indentation(line_start, view, tab_size, indentation_limit=None):
+  if indentation_limit is None:
+    row = view.rowcol(line_start)[0]
+    indentation_limit = view.text_point(row + 1, 0) - line_start
+
+  pos = 0
+  for ch in view.substr(sublime.Region(line_start, line_start + indentation_limit)):
     if ch == '\t':
       pos += tab_size - (pos % tab_size)
 
-    elif ch.isspace():
+    elif ch == ' ':
       pos += 1
 
     else:
       break
 
-    if limit <= pos:
+    if indentation_limit <= pos:
       break
 
   return pos
 
+
 def is_white_row(view, points):
   return all(view.substr(pt).isspace() for pt in reversed(points))
 
-def get_breadcrumb(view, points, limit):
-  for pt in points:
-    ch = view.substr(pt)
-    if not ch.isspace():
-      return view.substr(sublime.Region(pt, min(view.line(pt).b, pt + limit))).strip()
-  return ''
-
-class BreadcrumbsCommand(sublime_plugin.EventListener):
-  def on_selection_modified(self, view):
-    tab_size = get_tab_size(view)
-    settings = sublime.load_settings('Breadcrumbs.sublime-settings')
-    breadcrumb_length_limit = settings.get('breadcrumb_length_limit', 100)
-    separator = settings.get('breadcrumbs_separator', u' ■ ')
-    total_breadcrumbs_length_limit = settings.get('total_breadcrumbs_length_limit', 200)
-    view.erase_status('breadcrumbs')
 
-    if len(view.sel()) == 0:
-      return
+def get_separator(view):
+  defaults = sublime.load_settings('Breadcrumbs.sublime-settings')
+  default_separator = defaults.get('breadcrumbs_separator', u' ')
+  return view.settings().get('breadcrumbs_separator', default_separator)
 
-    current_row = view.rowcol(view.sel()[0].b)[0]
 
-    def get_row_start(row):
-      return view.text_point(row,0)
+def get_statusbar_enabled(settings):
+  defaults = sublime.load_settings('Breadcrumbs.sublime-settings')
+  default_enabled = defaults.get('show_breadcrumbs_in_statusbar', True)
+  return settings.get('show_breadcrumbs_in_statusbar', default_enabled)
```

Comment:
It could be even DRYier, like:
```
def get_setting(view_settings, key, default_value):
  defaults = sublime.load_settings('Breadcrumbs.sublime-settings')
  return view_settings().get(key, defaults.get(key, default_value))

def get_statusbar_enabled(settings):
  return get_setting(settings, 'show_breadcrumbs_in_statusbar', True)

def get_separator(view):
  return get_setting(view.settings(), 'breadcrumbs_separator', u' ')

```
although I would reconsider if this is a good thing that the two functions differ in the type of the argument (view vs settings)

You could also reuse this pattern for the `breadcrumb_regex`.